apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/model/SimpleXAPI.kt

```diff
@@ -1549,111 +1552,152 @@ object ChatController {
     }
   }
 
-  suspend fun apiReceiveFiles(rh: Long?, fileIds: List<Long>, userApprovedRelays: Boolean, encrypted: Boolean, inline: Boolean? = null): List<Long> {
-    val failures = mutableListOf<Long>()
-    val serversToApprove = mutableSetOf<String>()
-    val filesIdsFromNonApprovedSources = mutableListOf<Long>()
+  suspend fun receiveFiles(rhId: Long?, user: UserLike, fileIds: List<Long>, userApprovedRelays: Boolean = false, auto: Boolean = false) {
+    val filesAccepted = mutableListOf<AChatItem>()
+    val fileIdsToApprove = mutableStateListOf<Long>()
+    val srvsToApprove = mutableSetOf<String>()
+    val otherFileErrs = mutableListOf<CR>()
 
-    fileIds.forEach { fileId ->
-      val r = sendCmd(rh, CC.ReceiveFile(fileId, userApprovedRelays = userApprovedRelays, encrypt = encrypted, inline = inline))
+    for (fileId in fileIds) {
+      val r = sendCmd(
+        rhId, CC.ReceiveFile(
+          fileId,
+          userApprovedRelays = userApprovedRelays,
+          encrypt = appPrefs.privacyEncryptLocalFiles.get(),
+          inline = null
+        )
+      )
+      if (r is CR.RcvFileAccepted) {
+        filesAccepted.add(r.chatItem)
+      } else {
+        val maybeChatError = chatError(r)
+        if (maybeChatError is ChatErrorType.FileNotApproved) {
+          fileIdsToApprove.add(maybeChatError.fileId)
+          srvsToApprove.addAll(maybeChatError.unknownServers.map { serverHostname(it) })
+        } else {
+          otherFileErrs.add(r)
+        }
+      }
+    }
 
-      when (r) {
-        is CR.RcvFileAcceptedSndCancelled -> failures.add(fileId)
-        is CR.RcvFileCancelled -> failures.add(fileId)
-        is CR.RcvFileSndCancelled -> failures.add(fileId)
-        else -> {
-          if (!(networkErrorAlert(r))) {
-            when (val maybeChatError = chatError(r)) {
-              is ChatErrorType.FileAlreadyReceiving -> {}
-              is ChatErrorType.FileNotApproved -> {
-                if (userApprovedRelays) {
-                  failures.add(fileId)
-                } else {
-                  serversToApprove.addAll(maybeChatError.unknownServers.map { serverHostname(it) })
-                  filesIdsFromNonApprovedSources.add(fileId)
-                }
-              }
+    filesAccepted.forEach {
+      chatItemSimpleUpdate(rhId, user, it)
+    }
 
-              else -> {
-                failures.add(fileId)
-              }
+    if (!auto) {
+      // If there are not approved files, alert is shown the same way both in case of singular and plural files reception
+      if (fileIdsToApprove.isNotEmpty()) {
+        showFilesToApproveAlert(
+          srvsToApprove = srvsToApprove.toList(),
```

Comment:
No reason to make .toList() here and on the next line. It can be as is without toList

Reply:
this one's set, second is unnecessary yes